backend/caldav/caldav.php

```diff
@@ -82,7 +82,11 @@ public function BackendCalDAV() {
      */
     public function Logon($username, $domain, $password) {
         $this->_caldav_path = str_replace('%u', $username, CALDAV_PATH);
-        $this->_caldav = new CalDAVClient(CALDAV_SERVER . ":" . CALDAV_PORT . $this->_caldav_path, $username, $password);
+	
+	// If domain is empty we use username's domain
+	$domain_replace = ( $domain ) ? $domain : explode('@', $username,2)[1];
```

Comment:
If $username doesn't contains '@' this will fail. For example, with the default config (USE_FULLEMAIL_FOR_LOGIN = false), or if the username doesn't contain the domain part (single domain server).

You should check if it contains '@', so it could be reduced to this:
```
strpos($domain, '@') === false ? $domain : explode('@', $username, 2)[1];
```